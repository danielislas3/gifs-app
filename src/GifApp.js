import React from 'react';
import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({defaultCategories=[]}) => {
  const [categories, setcategories] = useState(defaultCategories);

  return (
    <>
      <h1>GifApp</h1>
      <AddCategory setCategories={setcategories} />
      <hr />

      <ol>
        {categories.map((e, i) => (
          <GifGrid category={e} key={e} />
        ))}
      </ol>
    </>
  );
};
