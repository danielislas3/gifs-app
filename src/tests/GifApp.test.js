import React from 'react';
import { shallow } from 'enzyme';
import { GifApp } from '../GifApp';
import '@testing-library/jest-dom';

describe('Pruebas en >GifApp/>', () => {

  test('Debe mostrar el componente', () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    

    const catregories=['one punch', 'gragon ball']
    const wrapper= shallow(<GifApp defaultCategories={catregories}/>)

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(catregories.length)
  })
  
});
