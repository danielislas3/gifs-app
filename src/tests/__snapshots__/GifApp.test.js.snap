// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en >GifApp/> Debe mostrar el componente 1`] = `
<Fragment>
  <h1>
    GifApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en >GifApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="gragon ball"
      key="gragon ball"
    />
  </ol>
</Fragment>
`;
